src/greyVest/NestedPicker.js

```diff
@@ -24,8 +24,54 @@ let getItemLabel = item =>
     ? F.cascade(['shortLabel', 'label'], item)
     : _.startCase(item._key)
 
+/*
+  if field "path" is a nested object with a single field (no other field has the same parent path),
+  the path is transformed to combine the object name and the field name into a new field
+*/
+let rollupSingleFieldObjects = _.flow(
```

Comment:
Would really appreciate some input/feedback how to make this more concise and user-friendly. The idea is to re-write the `path` field when we figure out we have a single field in the field group (and it should keep rolling up until we are either at root or we find a common parent further up in the path hierarchy). Then re-process the fields until no more paths are modified.

`Trees are coming`